### PR TITLE
Improved comment-type handling

### PR DESCRIPTION
**Pull request by pfefferle**
It now uses the comment-type as index and keeps a list of mapped activity-types.

Also added a function to get the comment-type by the activity-type.

`Comment::get_comment_type_by_activity_type `

props @obenland

### CODE REVIEW

File: includes/class-comment.php [hunk 1]
```diff
@@ -523,8 +545,8 @@ public static function get_comment_types() {
 	 * @return boolean True if registered.
 	 */
 	public static function is_registered_comment_type( $slug ) {
-		$slug = strtolower( $slug );
-		$slug = sanitize_key( $slug );
+		$slug = \strtolower( $slug );
+		$slug = \sanitize_key( $slug );
 
 		return in_array( $slug, array_keys( self::get_comment_types() ), true );
```
Comment (obenland) [inline includes/class-comment.php]:
```suggestion
		$registered_comment_types = self::get_comment_types();

		return isset( $registered_comment_types[ $slug ] );
```
Would this work? I wonder if that's faster.

**Comment by obenland** [comment]
With this change, `get_comment_type_names()` could be deprecated in favor of  `get_comment_type_slugs()` and just return `array_keys( self::get_comment_types() )`

File: includes/class-comment.php [hunk 2]
```diff
@@ -520,14 +542,17 @@ public static function get_comment_types() {
 	/**
 	 * Is this a registered comment type.
 	 *
-	 * @param string $slug The name of the type.
+	 * @param string $name The name of the type.
```
Comment (obenland) [inline includes/class-comment.php]: I'd leave it as slug, as that's what it is (hence my proposal to rename `get_comment_type_names()` to `get_comment_type_slugs()`). The name would be the translated "Repost" and "Like".

Comment (pfefferle) [reply includes/class-comment.php]: so reverting this commit?

Comment (obenland) [reply includes/class-comment.php]: oh, I didn't realized you had changed it and reverted again. Yes, I think so, if you don't have a strong preference. [WP uses slug for these things](https://developer.wordpress.org/?s=slug).

File: includes/collection/class-interactions.php [hunk 3]
```diff
@@ -111,14 +111,13 @@ public static function add_reaction( $activity ) {
 			return false;
 		}
 
-		$type = $activity['type'];
+		$comment_type = Comment::get_comment_type_by_activity_type( $activity['type'] );
```
Comment (obenland) [inline includes/collection/class-interactions.php:114]: Could there ever be multiple comment types for an activity type? Is that something we have to worry about?

**Review by obenland — APPROVED** [review]

File: includes/collection/class-interactions.php (hunk 3 above)
Comment (pfefferle) [reply includes/collection/class-interactions.php:114]: I don't think so... there is the possibility that we might store items in different places (direct messages for example), but that is a different discussion...

**Pull request merged**